src/flint/fmpz.jl

```diff
@@ -1284,10 +1288,22 @@ function primorial(x::Int)
     x < 0 && throw(DomainError("Argument must be non-negative: $x"))
     z = fmpz()
     ccall((:fmpz_primorial, :libflint), Nothing,
-          (Ref{fmpz}, UInt), z, x)
-    return z
+          (Ref{fmpz}, UInt), z, UInt(x))
+    return Int(z)
 end
 
+@doc Markdown.doc"""
+    primorial(x::fmpz)
+>  Return the primorial of $x$, i.e. the product of all primes less than or
+> equal to $x$. If $x < 0$ we throw a `DomainError()`.
+"""
+function primorial(x::fmpz)
```

Comment:
I suggest merging the two implementations: the signature can be `primorial(x::Union{Int,fmpz})`, and the last line `return convert(typeof(x), z)`.
Similar for other functions below. If you don't want to bother now but are fine in principle, I can do it in a subsequent PR.

Reply:
Yeah I thought you would suggest that. I'll leave it for you to sort out.